src/main/java/org/example/Multi.java

```diff
@@ -0,0 +1,104 @@
+package org.example;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+
+public class Multi {
+
+    private ExecutorService executorService = Executors.newFixedThreadPool(3);
+
+    public void start1() {
+        Future<String> future1 = executorService.submit(createTask("1"));
+        Future<String> future2 = executorService.submit(createTask("2"));
+        Future<String> future3 = executorService.submit(createTask("3"));
+
+        try {
+          String res1 =  future1.get();
+          String res2 =  future2.get();
+          String res3 =  future3.get();
+          System.out.println(res1);
+          System.out.println(res2);
+          System.out.println(res3);
+        } catch (Exception err) {
+            System.out.println(err);
+        }
+        executorService.shutdown();
+    }
+
+    public void start2() {
+        final List<Callable<String>> callables = IntStream.rangeClosed(1, 3)
```

Comment:
Just as an experiment for yourself, try to create 1000 or 10 000 tasks with the corresponding amount of threads. If it works for 1 sec, try to find the approximate number, when it crashes. 
LMK if you find this number.

Then you can try to change executor to newWorkStealingPool and newCachedThreadPool. And see how it works again with huge amount of tasks 10k or more.

Again, lmk your results

Reply:
i checked, when work has more than 1 sec:

With newFixedThreadPool(3):
start2() method works slow when number of tasks more or equal 4 
start4() method works slow when number of tasks more or equal 6

With newWorkStealingPool():
start2() method works slow when number of tasks more or equal 7
start4() method works slow when number of tasks more or equal 6
but if change level of parallelism (for example 1000), we can execute more than 1000 tasks

With newCachedThreadPool():
start2() method works slow when number of tasks more or equal 6000
start4() method works slow when number of tasks more or equal 6

Reply:
of course if you increase the amount of tasks with fixed 3 threads you will get problems when tasks more than 3.
You have to increase amount of threads as well.
If tasks = 1k then set amount of threads 1k as well. 
btw complableFutures use executor service under the hood.